test/test_spath.c

```diff
@@ -0,0 +1,361 @@
+  if(rc != SPATH_FAILURE){
+    printf ("Error in line %d, file %s, function %s.\n", __LINE__, __FILE__, __func__);
+    printf("test_spath failed, spath_insert_str() attempt to insert into NULL spath didn't return failure\n");
+    return TEST_FAIL;
+  }
+
+  spath_delete(&sp);
+  sp = spath_from_str("Directory");
+  spath_append_str(sp, "File");
+  spath_strcpy(buff,100, sp);
+  if(strcmp(buff, "Directory/File") != 0 ){
+    printf ("Error in line %d, file %s, function %s.\n", __LINE__, __FILE__, __func__);
+    printf("test_spath failed, spath_append_str() failed \n");
+    return TEST_FAIL;
+  }
+
+  spath_delete(&sp);
+  rc = spath_append_str(sp,"File");
+  if(rc != SPATH_FAILURE){
+    printf ("Error in line %d, file %s, function %s.\n", __LINE__, __FILE__, __func__);
+    printf("test_spath failed, spath_append_str() attempt to append to NULL spath didn't return failure\n");
+    return TEST_FAIL;
+  }
+
+  spath_delete(&sp);
+  sp = spath_from_str("File");
+  spath_prepend_str(sp, "Directory");
+  spath_strcpy(buff,100, sp);
+  if(strcmp(buff, "Directory/File") != 0 ){
+    printf ("Error in line %d, file %s, function %s.\n", __LINE__, __FILE__, __func__);
+    printf("test_spath failed, spath_prepend_str() failed \n");
+    return TEST_FAIL;
+  }
+
+  spath_delete(&sp);
+  rc = spath_prepend_str(sp,"Directory");
+  if(rc != SPATH_FAILURE){
+    printf ("Error in line %d, file %s, function %s.\n", __LINE__, __FILE__, __func__);
+    printf("test_spath failed, spath_prepand_str() attempt to prepend to NULL spath didn't return failure\n");
+    return TEST_FAIL;
+  }
+
+  spath_delete(&sp);
+  sp = spath_from_str("Directory/File/1/2/3/4");
+  rc = spath_slice(sp, 2,4);
+//  spath_slice(sp, -6,-17);
+  spath_strcpy(buff,100, sp);
+  printf("slice result is %s\n",buff);
+  if(rc != SPATH_SUCCESS || strcmp(buff, "1/2/3/4") != 0 ){
+    printf ("Error in line %d, file %s, function %s.\n", __LINE__, __FILE__, __func__);
+    printf("test_spath failed, spath_slice() failed \n");
+    return TEST_FAIL;
+  }
+
+//This test has been invalidated by Adam Moody's changes to functionality in Pull Request 6 on Github.
+//  spath_delete(&sp);
+//  sp = spath_from_str("Directory/File/1/2/3/4");
+//  rc = spath_slice(sp,2,5);
+//  if(rc == SPATH_SUCCESS){
+//    printf ("Error in line %d, file %s, function %s.\n", __LINE__, __FILE__, __func__);
+//    printf("test_spath failed, spath_slice() returned SPATH_SUCCESS when (offset + length) > components \n");
+//    return TEST_FAIL;
+//  }
+
+//This test has been invalidated by Adam Moody's changes to functionality in Pull Request 6 on Github.
+//  spath_delete(&sp);
+//  sp = spath_from_str("Directory/File/1/2/3/4");
+//  rc = spath_slice(sp,6,1);
+//  if(rc == SPATH_SUCCESS){
+//    printf ("Error in line %d, file %s, function %s.\n", __LINE__, __FILE__, __func__);
+//    printf("test_spath failed, spath_slice() returned SPATH_SUCCESS when starting index not in range \n");
+//    return TEST_FAIL;
+//  }
```

Comment:
We can also update this test to match expected behavior.  I think this should result in a string that is "Directory".